flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1373,6 +1373,7 @@ private void cleanupAfterCompletedCheckpoint(
                     completedCheckpoint.getTimestamp(),
                     extractIdIfDiscardedOnSubsumed(lastSubsumed));
         }
+        pendingCheckpoint.getCompletionFuture().complete(completedCheckpoint);
```

Comment:
If we encounter an exception in `addCompletedCheckpointToStoreAndSubsumeOldest `, what should we do with `pendingCheckpoint. getCompleteFuture()`?